pyuvdata/cst_beam.py

```diff
@@ -0,0 +1,223 @@
+import os
+import sys
+import re
+import numpy as np
+from uvbeam import UVBeam
+import utils as uvutils
+
+
+class CSTBeam(UVBeam):
+    """
+    Defines a CST-specific subclass of UVBeam for reading CST text files.
+    This class should not be interacted with directly, instead use the
+    read_cst_beam method on the UVBeam class.
+
+    Assumes the structure in the simulation was symmetric under
+    45 degree rotations about the z-axis.
```

Comment:
Yes, it's inherent to CST because the CST files are so simple that they don't specify all the required metadata. But I'll add it as a parameter that defaults to 'x' but can be specified differently. Actually, it should support passing different files for the different pols... I'll think about that.